server/api/auth.go

```diff
@@ -19,123 +17,16 @@ import (
-	// required: true
-	Type string `json:"type"`
-
-	// If specified, login using username
-	// required: false
-	Username string `json:"username"`
-
-	// If specified, login using email
-	// required: false
-	Email string `json:"email"`
-
-	// Password
-	// required: true
-	Password string `json:"password"`
-
-	// MFA token
-	// required: false
-	// swagger:ignore
-	MfaToken string `json:"mfa_token"`
-}
-
-// LoginResponse is a login response
-// swagger:model
-type LoginResponse struct {
-	// Session token
-	// required: true
-	Token string `json:"token"`
-}
-
-func LoginResponseFromJSON(data io.Reader) (*LoginResponse, error) {
-	var resp LoginResponse
-	if err := json.NewDecoder(data).Decode(&resp); err != nil {
-		return nil, err
-	}
-	return &resp, nil
-}
-
-// RegisterRequest is a user registration request
-// swagger:model
-type RegisterRequest struct {
-	// User name
-	// required: true
-	Username string `json:"username"`
-
-	// User's email
-	// required: true
-	Email string `json:"email"`
-
-	// Password
-	// required: true
-	Password string `json:"password"`
-
-	// Registration authorization token
-	// required: true
-	Token string `json:"token"`
-}
-
-func (rd *RegisterRequest) IsValid() error {
-	if strings.TrimSpace(rd.Username) == "" {
-		return ParamError{"username is required"}
-	}
-	if strings.TrimSpace(rd.Email) == "" {
-		return ParamError{"email is required"}
+func (a *API) registerAuthRoutes(r *mux.Router) {
+	// personal-server specific routes. These are not needed in plugin mode.
+	if !a.isPlugin {
+		r.HandleFunc("/login", a.handleLogin).Methods("POST")
+		r.HandleFunc("/logout", a.sessionRequired(a.handleLogout)).Methods("POST")
+		r.HandleFunc("/register", a.handleRegister).Methods("POST")
+		r.HandleFunc("/teams/{teamID}/regenerate_signup_token", a.sessionRequired(a.handlePostTeamRegenerateSignupToken)).Methods("POST")
 	}
-	if !auth.IsEmailValid(rd.Email) {
-		return ParamError{"invalid email format"}
-	}
-	if rd.Password == "" {
-		return ParamError{"password is required"}
-	}
-	return isValidPassword(rd.Password)
-}
-
-// ChangePasswordRequest is a user password change request
-// swagger:model
-type ChangePasswordRequest struct {
-	// Old password
-	// required: true
-	OldPassword string `json:"oldPassword"`
 
-	// New password
-	// required: true
-	NewPassword string `json:"newPassword"`
-}
-
-// IsValid validates a password change request.
-func (rd *ChangePasswordRequest) IsValid() error {
-	if rd.OldPassword == "" {
-		return ParamError{"old password is required"}
-	}
-	if rd.NewPassword == "" {
-		return ParamError{"new password is required"}
-	}
-	return isValidPassword(rd.NewPassword)
-}
-
-func isValidPassword(password string) error {
-	if len(password) < MinimumPasswordLength {
-		return ParamError{fmt.Sprintf("password must be at least %d characters", MinimumPasswordLength)}
-	}
-	return nil
+	r.HandleFunc("/users/{userID}/changepassword", a.sessionRequired(a.handleChangePassword)).Methods("POST")
```

Comment:
This call is only allow if it is a plugin, so... probably we can move it inside the `isPlugin` block, clearly... it is not something that needs to be in this PR. actually I prefer a different one for that, because this is already huge change of restructuring.

Reply:
I have made note of this.  Thanks!